core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedValueResolver.java

```diff
@@ -675,7 +678,14 @@ private static AnnotatedValueResolver ofInjectableTypes0(AnnotatedElement annota
 
                 // Do not convert value here because the element type is String.
                 if (values != null && !values.isEmpty()) {
-                    values.stream().map(resolver::convert).forEach(resolvedValues::add);
+                    if (values.size() == 1) {
```

Comment:
Question: What happen if a client sends `params=1,2,3&params=4,5,6`? Should it be converted to `List(1,2,3,4,5,6)` or `List("1,2,3", "4,5,6")`?

Reply:
As multiple `params`s are used, `List("1,2,3", "4,5,6")` makes sense.

>  Would it be better to make comma-delimited optional?

I'm not sure. Even though it is not a standard protocol, it is widely used for many libraries. I am OK to enable it by default. 
@trustin @jrhee17 Any thought on this?

Reply:
How do others do such cases?

Reply:
For the following curl:
```
curl -v "http://localhost:8080?params=1,2,3"
curl -v "http://localhost:8080?params=1,2,3&params=4,5,6"
```

### Spring
API:
```
@GetMapping("/")
public ResponseEntity<String> multiParams(@RequestParam("params") List<String> params) {
    return ResponseEntity.ok(String.join(":", params));
}
```
Received:
```
1:2:3
1,2,3:4,5,6
```

Let me check other frameworks soon.

Reply:
> Let me check other frameworks soon.

I don't think we don't have to check other frameworks. 😄 
Let's approach in the same direction.